collectors/proc.plugin/proc_net_wireless.c

```diff
@@ -0,0 +1,452 @@
+
+            if (previous) {
+                previous->next = current;
+            } else {
+                netwireless_root = current;
+            }
+        } else {
+            previous = current;
+            current = current->next;
+        }
+    }
+}
+
+// finds an existing interface or creates a new entry
+static struct netwireless *find_or_create_wireless(const char *name)
+{
+    struct netwireless *wireless;
+    uint32_t hash = simple_hash(name);
+
+    // search it, from begining to the end
+    for (wireless = netwireless_root ; wireless ; wireless = wireless->next) {
+        if (unlikely(hash == wireless->hash && !strcmp(name, wireless->name))) {
+            return wireless;
+        }
+    }
+
+    // create a new one
+    wireless = callocz(1, sizeof(struct netwireless));
+    wireless->name = strdupz(name);
+    wireless->hash = hash;
+
+    // link it to the end
+    if (netwireless_root) {
+        struct netwireless *last_node;
+        for (last_node = netwireless_root; last_node->next ; last_node = last_node->next);
+
+        last_node->next = wireless;
+    } else
+        netwireless_root = wireless;
+
+    return wireless;
+}
+
+static void configure_device(int do_status, int do_quality, int do_discarded_packets, int do_missed,
+                             struct netwireless *wireless_dev) {
+    wireless_dev->do_status = do_status;
+    wireless_dev->do_quality = do_quality;
+    wireless_dev->do_discarded_packets = do_discarded_packets;
+    wireless_dev->do_missed_beacon = do_missed;
+    wireless_dev->configured = true;
+
+    char buffer[RRD_ID_LENGTH_MAX + 1];
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_status", wireless_dev->name);
+    wireless_dev->chart_id_net_status = strdupz(buffer);
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_link", wireless_dev->name);
+    wireless_dev->chart_id_net_link = strdupz(buffer);
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_level", wireless_dev->name);
+    wireless_dev->chart_id_net_level = strdupz(buffer);
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_noise", wireless_dev->name);
+    wireless_dev->chart_id_net_noise = strdupz(buffer);
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_discarded_packets", wireless_dev->name);
+    wireless_dev->chart_id_net_discarded_packets = strdupz(buffer);
+
+    snprintfz(buffer, RRD_ID_LENGTH_MAX, "%s_missed_beacon", wireless_dev->name);
+    wireless_dev->chart_id_net_missed_beacon = strdupz(buffer);
+}
+
+int do_proc_net_wireless(int update_every, usec_t dt)
+{
+    UNUSED(dt);
+    static procfile *ff = NULL;
+    static int do_status, do_quality = -1, do_discarded_packets, do_beacon;
+    static char *proc_net_wireless_filename = NULL;
+
+    if (unlikely(do_quality == -1)) {
+        char filename[FILENAME_MAX + 1];
+        snprintfz(filename, FILENAME_MAX, "%s%s", netdata_configured_host_prefix, "/proc/net/wireless");
+
+        proc_net_wireless_filename = strdupz(filename);
```

Comment:
Why have you removed a configuration option for the file?

Reply:
The issue #6333 was explicit about the file `/proc/net/wireless` and when see the comments about this file inside the [kernel source](https://elixir.bootlin.com/linux/latest/source/net/wireless/wext-proc.c), it looks like there is not motive to change it.

Reply:
Usually, we make such files configurable. It's also convenient for testing.

Reply:
> It's also convenient for testing.

i was about to test it and realized there is no way to do it 😢 Wanted to test add/remove interfaces in runtime

Reply:
I brought it back.